Rhinoceros_Engine/Query/DocumentTolerance.cs

```diff
@@ -0,0 +1,49 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Rhino;
+
+namespace BH.Engine.Rhinoceros
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        public static double DocumentTolerance()
+        {
+            RhinoDoc doc = Rhino.RhinoDoc.ActiveDoc;
+
+            if (doc != null)
+                return doc.ModelAbsoluteTolerance;
+            else
+            {
+                BH.Engine.Reflection.Compute.RecordError("Rhino document tolerance could not be retrieved because active document has not been found.");
+                return double.NaN;
```

Comment:
I am not sure returning something that is not a number is useful here.
This forces checks in downstream methods whether the tolerance is a valid signal or not.
`NaN` might be especially bad.

Is there a reason you are not returning the standard BHoM tolerance (maybe with a warning) when the doc tolerance fails?

Reply:
Yeah, I was thinking about it too, but decided to record the error to inform the user that something went badly wrong instead of hiding the problem under a default value. However, I am 50/50 on that, can see the benefits of warning + returning the standard BHoM tolerance.

@al-fisher @LMarkowski how does that fit into the wider discussion about returning doubles on fail?

Reply:
At this moment every not implemented case in the Geometry_Engine will return `double.NaN`. I'm not sure if it's to be expanded on every other error but to me that sounds reasonable.

Reply:
guess this is a question of returning something that for sure will make things fail and raise an error (current implementation) or return something that for many/most cases will work, but for some cases might not and raise a warning (@epignatelli suggestion).

Would probably err on the side of eduardos suggestion for this case. Personally think it is a slightly different context/situation compared to a method not being implemented in the Geometry_Engine, but good to get @al-fisher s oppinion on the matter as well.

Reply:
Yeah, I think the problem is that we are using our own logging mechanism to avoid throwing an exception, but we are returning `NaN`. The code will break somewhere else due to the `NaN`s. so that's not particularly useful.

Reply:
Yes - agreed - think this might be a sensible case to raise a warning and returning the standard BHoM tolerance.
With message explaining "no active document found - default BHoM tolerance returned."

Reply:
Changes made, ready for review 👍